### PR TITLE
Closes #354: Extract converters from MatchableEntity

### LINKED ISSUE #354: Remove the coupling between MatchableEntity and external conansys modules

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/BasicMetadataToEntityConverterTest.scala [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+ */
+class BasicMetadataToEntityConverterTest {
+  
+  val basicMetadataConverter = new BasicMetadataToEntityConverter() 
+  
+  
+  @Test(groups = Array("fast"))
+  def convertTest() {
+    
+    val basicMetadata = BasicMetadata.newBuilder()
```
Comment (lukdumi) [inline citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/BasicMetadataToEntityConverterTest.scala]: given/execute/assert ;-)

File: citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/BasicMetadataToEntityConverterTest.scala [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    
+    val expectedEntity = MatchableEntity.fromParametersExt("some_id", "John Doe", "Some Journal", "Some Title", "12-19", "2001", "23", "345", null)
+    assertMatchableEntityEquals(actualEntity, expectedEntity)
+    
+  }
+  
+  
+  @Test(groups = Array("fast"))
+  def convertEmptyTest() {
+
```
Comment (lukdumi) [inline citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/BasicMetadataToEntityConverterTest.scala]: same here and below

File: citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/DocumentMetadataToEntityConverterTest.scala [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  
+  @BeforeMethod
+  def setup() {
+    reset(basicMetadataConverter)
+  }
+  
+  
+  @Test(groups = Array("fast"))
+  def convertTest() {
+
```
Comment (lukdumi) [inline citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/DocumentMetadataToEntityConverterTest.scala]: given/execute/assert

File: citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/RawReferenceToEntityConverterTest.scala [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  @BeforeMethod
+  def setup() {
+    reset(referenceParser)
+  }
+  
+  
+  @Test(groups = Array("fast"))
+  def convertTest() {
+    
+    val bibEntry = new BibEntry()
```
Comment (lukdumi) [inline citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/RawReferenceToEntityConverterTest.scala]: given/execute/assert

File: citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/BasicMetadataToEntityConverterTest.scala (hunk 1 above)
Comment (madryk) [inline citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/BasicMetadataToEntityConverterTest.scala]: ok

File: citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/BasicMetadataToEntityConverterTest.scala (hunk 2 above)
Comment (madryk) [inline citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/BasicMetadataToEntityConverterTest.scala]: ok

File: citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/DocumentMetadataToEntityConverterTest.scala (hunk 3 above)
Comment (madryk) [inline citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/DocumentMetadataToEntityConverterTest.scala]: ok

File: citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/RawReferenceToEntityConverterTest.scala (hunk 4 above)
Comment (madryk) [inline citation-matching/citation-matching-coansys-code/src/test/scala/pl/edu/icm/coansys/citations/converters/RawReferenceToEntityConverterTest.scala]: ok

**Pull request closed without merging**